xtable-service/src/main/java/org/apache/xtable/service/ConversionService.java

```diff
@@ -0,0 +1,156 @@
+
+  /**
+   * Constructs a ConversionService instance with required dependencies.
+   *
+   * @param sparkHolder the Spark holder instance containing the Spark context and configuration
+   * @param conversionServiceUtil utility for handling metadata operations
+   */
+  @Inject
+  public ConversionService(SparkHolder sparkHolder, ConversionServiceUtil conversionServiceUtil) {
+    this.sparkHolder = sparkHolder;
+    this.conversionServiceUtil = conversionServiceUtil;
+    this.conversionController = new ConversionController(sparkHolder.jsc().hadoopConfiguration());
+  }
+
+  /**
+   * Constructs a ConversionService instance using dependency injection for testing.
+   *
+   * @param sparkHolder the Spark holder instance
+   * @param conversionController a preconfigured conversion controller
+   * @param conversionServiceUtil utility for handling metadata operations
+   */
+  @VisibleForTesting
+  public ConversionService(
+      SparkHolder sparkHolder,
+      ConversionServiceUtil conversionServiceUtil,
+      ConversionController conversionController) {
+    this.sparkHolder = sparkHolder;
+    this.conversionController = conversionController;
+    this.conversionServiceUtil = conversionServiceUtil;
+  }
+
+  /**
+   * Converts a source table to one or more target table formats.
+   *
+   * <p>The method builds a SourceTable based on the request parameters and constructs corresponding
+   * TargetTable instances for each target format. It then performs a synchronous conversion through
+   * the conversion controller. After conversion, it retrieves schema and metadata paths for each
+   * target table.
+   *
+   * @param convertTableRequest the conversion request containing source table details and target
+   *     formats
+   * @return a response containing details of converted target tables
+   */
+  public ConvertTableResponse convertTable(ConvertTableRequest convertTableRequest) {
+    SourceTable sourceTable =
+        SourceTable.builder()
+            .name(convertTableRequest.getSourceTableName())
+            .basePath(convertTableRequest.getSourceTablePath())
+            .formatName(convertTableRequest.getSourceFormat())
+            .build();
+
+    List<TargetTable> targetTables = new ArrayList<>();
+    for (String targetFormat : convertTableRequest.getTargetFormats()) {
+      TargetTable targetTable =
+          TargetTable.builder()
+              .name(convertTableRequest.getSourceTableName())
+              .basePath(convertTableRequest.getSourceTablePath())
+              .formatName(targetFormat)
+              .build();
+      targetTables.add(targetTable);
+    }
+
+    ConversionConfig conversionConfig =
+        ConversionConfig.builder().sourceTable(sourceTable).targetTables(targetTables).build();
+
+    ConversionSourceProvider<?> conversionSourceProvider =
+        conversionServiceUtil.getConversionSourceProvider(
+            convertTableRequest.getSourceFormat(), sparkHolder.jsc().hadoopConfiguration());
```

Comment:
You can do this and avoid the class `ConversionSourceUtil` class completely.
  `conversionSourceProvider.getConversionSourceInstance(sourceTable).getCurrentTable() `

Reply:
Will try refactoring to what you suggested.

Reply:
So the full context of the change is the following

```
ConversionSourceProvider<?> conversionSourceProvider =
        conversionServiceUtil.getConversionSourceProvider(
            convertTableRequest.getSourceFormat(), sparkHolder.jsc().hadoopConfiguration());
    conversionController.sync(conversionConfig, conversionSourceProvider);
```

Im not sure how the line you shared `conversionSourceProvider.getConversionSourceInstance(sourceTable).getCurrentTable()` helps, as the `conversionController.sync` is looking for a `ConversionSourceProvider` which is what the util is provding.

Reply:
Discussed with Vinish offline